packages/web/src/components/spa/HackAUBG/RegistrationForm/RegistrationForm.jsx

```diff
@@ -115,229 +109,233 @@ const RegistrationForm = () => {
-                    <div className={styles.form_row}>
-                        <InputComponent
-                            label="First Name*"
-                            type="name"
-                            required="true"
-                            register={register}
-                            name="first_name"
-                            error={errors.first_name && errors.first_name.message}
-                        />
-                        <InputComponent
-                            label="Last Name*"
-                            type="name"
-                            required="true"
-                            register={register}
-                            name="last_name"
-                            error={errors.last_name && errors.last_name.message}
-                        />
-                    </div>
-                    <div className={styles.form_row}>
-                        <InputComponent
-                            label="Email*"
-                            type="email"
-                            required="true"
-                            register={register}
-                            name="email"
-                            error={errors.email && errors.email.message}
-                        />
-                        <InputComponent
-                            label="Age*"
-                            type="age"
-                            required="true"
-                            register={register}
-                            name="age"
-                            error={errors.age && errors.age.message}
-                        />
-                    </div>
-                    <div className={styles.form_row}>
-                        <InputComponent
-                            label="Location*"
-                            type="text"
-                            required="Field is required"
-                            register={register}
-                            name="location"
-                            error={errors.location && errors.location.message}
-                        />
-                        <InputComponent
-                            label="Choose a School/University*"
-                            type="select"
-                            required="Field is required"
-                            register={register}
-                            values={[
-                                'American University in Bulgaria',
-                                'Sofia University',
-                                'Technical University - Sofia',
-                                'Plovdiv University',
-                                'Other',
-                            ]}
-                            name="university"
-                            error={errors.university && errors.university.message}
-                        />
+    if (featureSwitches.regForm) {
+        if ((featureSwitches.isRegFull === false && isApiUp) || (teamName && isApiUp)) {
```

Comment:
That's some pretty wild conditional statements.

Reply:
I believe this could be rewritten to `isApiUp && (!featureSwitches.isRegFull || teamName)`

Reply:
yup it can be